### PR TITLE
Autonomous integration

### PR DESCRIPTION
**Pull request by Andor16**
I have updated the R1, R2, and B2 programs to handle grabbing a block at the beginning of the game and taking it to one of the cryptoboxes. These programs do work, although they have not yet been very thoroughly tested. They simply need to go through the 3 test process.

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B1.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    Servo rightArm;
+    ColorSensor colorSensorLeft;
+    DistanceSensor distanceSensorLeft;
+
+    //OpenGLMatrix lastLocation = null;
+    //VuforiaLocalizer vuforia;
+
+    //VuforiaTrackables relicTrackables = this.vuforia.loadTrackablesFromAsset("RelicVuMark");
+    //VuforiaTrackable relicTemplate = relicTrackables.get(0);
+    //RelicRecoveryVuMark vuMark;
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B1.java]: #Fix Remove all the unused Vuforia code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B1.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+        int cameraMonitorViewId = hardwareMap.appContext.getResources().getIdentifier("cameraMonitorViewId", "id", hardwareMap.appContext.getPackageName());
+        VuforiaLocalizer.Parameters parameters = new VuforiaLocalizer.Parameters(cameraMonitorViewId);
+
+        parameters.vuforiaLicenseKey = "AUkgO0T/////AAAAGaYeMjdF+Us8tdP9fJcRhP9239Bwgzo0STjrR4II0s58wT/ja6GlSAQi/ptpHERhBhdNq8MMmlxC6bjyebsGnr/26IxYKhFFdC67Q7HE0jhDrsrEfxfJMFnsk2zSdt5ofwm2Z1xNhdBg2kfFCzdodI7aHFEdUQ6fddoTioTSPu9zzU9XqBr7Ra+5mTaIwp10heZmlXIjWfu8220ef/tZQ8QSmDX1GSqRLBjUJspesff8Nv9pkQAK3Nvp8YFHKJoFNkSV7QJW7mi/liHYq6DxYqhWk977WYGwzhHA003HNV4OUWhTLJGiPsiFhAlcJVbnVMn6ldnsSauT4unjXA9VBIzaYtSJc29UJYmWyin3MxPz";
+
+        parameters.cameraDirection = VuforiaLocalizer.CameraDirection.FRONT;
+        //this.vuforia = ClassFactory.createVuforiaLocalizer(parameters);
+
+        //relicTemplate.setName("relicVuMarkTemplate");
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B1.java]: #Fix remove 69-77

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B1.java [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        goalPosition = 0.35;
+        position = 0.85;
+        leftArm.setPosition(position); //vertical start
+        rightArm.setPosition(0);
+        // position
+
+        state = States.TIME;
+        isFinished = false;
+
+        //relicTrackables.activate();
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B1.java]: #Fix remove

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B1.java [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+    }
+        @Override
+        public void loop(){
+            //telemetry.addData("Pitch:", robot.getPitch());
+            //telemetry.update();
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B1.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B1.java [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+            switch(state){
+                /*case SCAN:
+                    //Scans the pictograph to get correct column
+                    vuMark = RelicRecoveryVuMark.from(relicTemplate);
+                    if(vuMark != RelicRecoveryVuMark.UNKNOWN){
+                        telemetry.addData("Column:", vuMark);
+                        telemetry.update();
+                        state = States.TIME;
+                    } break;*/
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B1.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+    Servo rightArm;
+    ColorSensor colorSensorLeft;
+    DistanceSensor distanceSensorLeft;
+
+    //OpenGLMatrix lastLocation = null;
+    //VuforiaLocalizer vuforia;
+
+    //VuforiaTrackables relicTrackables = this.vuforia.loadTrackablesFromAsset("RelicVuMark");
+    //VuforiaTrackable relicTemplate = relicTrackables.get(0);
+    //RelicRecoveryVuMark vuMark;
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java]: #Fix remove commented code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+
+        int cameraMonitorViewId = hardwareMap.appContext.getResources().getIdentifier("cameraMonitorViewId", "id", hardwareMap.appContext.getPackageName());
+        VuforiaLocalizer.Parameters parameters = new VuforiaLocalizer.Parameters(cameraMonitorViewId);
+
+        parameters.vuforiaLicenseKey = "AUkgO0T/////AAAAGaYeMjdF+Us8tdP9fJcRhP9239Bwgzo0STjrR4II0s58wT/ja6GlSAQi/ptpHERhBhdNq8MMmlxC6bjyebsGnr/26IxYKhFFdC67Q7HE0jhDrsrEfxfJMFnsk2zSdt5ofwm2Z1xNhdBg2kfFCzdodI7aHFEdUQ6fddoTioTSPu9zzU9XqBr7Ra+5mTaIwp10heZmlXIjWfu8220ef/tZQ8QSmDX1GSqRLBjUJspesff8Nv9pkQAK3Nvp8YFHKJoFNkSV7QJW7mi/liHYq6DxYqhWk977WYGwzhHA003HNV4OUWhTLJGiPsiFhAlcJVbnVMn6ldnsSauT4unjXA9VBIzaYtSJc29UJYmWyin3MxPz";
+
+        parameters.cameraDirection = VuforiaLocalizer.CameraDirection.FRONT;
+        //this.vuforia = ClassFactory.createVuforiaLocalizer(parameters);
+
+        //relicTemplate.setName("relicVuMarkTemplate");
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java]: #Fix remove lines 77-80

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+        goalPosition = 0.35;
+        position = 0.85;
+        leftArm.setPosition(position); //vertical start
+        rightArm.setPosition(0);
+        // position
+
+        state = BlueStates.TIME;
+        isFinished = false;
+
+        //relicTrackables.activate();
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java]: #Fix remove

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+            //telemetry.update();
+            switch(state){
+                /*case SCAN:
+                    //Scans the pictograph to get correct column
+                    vuMark = RelicRecoveryVuMark.from(relicTemplate);
+                    if(vuMark != RelicRecoveryVuMark.UNKNOWN){
+                        telemetry.addData("Column:", vuMark);
+                        telemetry.update();
+                        state = BlueStates.TIME;
+                    } break;*/
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+        isFinished = false;
+
+        //relicTrackables.activate();
+
+    }
+        @Override
+        public void loop(){
+            currentSeconds = time.seconds();
+            //telemetry.addData("Pitch:", robot.getPitch());
+            //telemetry.update();
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java [hunk 11]
```diff
@@ -0,0 +1,279 @@
...
+                    } break;*/
+
+                /*case DRIVEZONE:
+                    //Drives into the parking zone. UNTESTED/DEACTIVATED
+                    if(!isFinished){
+                        isFinished = robot.encoderDrive(DriveTrain.Direction.S, 0.25, 15);
+                    } else{
+                        isFinished = false;
+                        state = States.END; //TURNBOX
+                    } break;*/
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Blue/Auto_B2.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+
+    ColorSensor colorSensorRight;
+    DistanceSensor distanceSensorRight;
+
+    //OpenGLMatrix lastLocation = null;
+    //VuforiaLocalizer vuforia;
+
+    //VuforiaTrackables relicTrackables = this.vuforia.loadTrackablesFromAsset("RelicVuMark");
+    //VuforiaTrackable relicTemplate = relicTrackables.get(0);
+    //RelicRecoveryVuMark vuMark;
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+
+        int cameraMonitorViewId = hardwareMap.appContext.getResources().getIdentifier("cameraMonitorViewId", "id", hardwareMap.appContext.getPackageName());
+        VuforiaLocalizer.Parameters parameters = new VuforiaLocalizer.Parameters(cameraMonitorViewId);
+
+        parameters.vuforiaLicenseKey = "AUkgO0T/////AAAAGaYeMjdF+Us8tdP9fJcRhP9239Bwgzo0STjrR4II0s58wT/ja6GlSAQi/ptpHERhBhdNq8MMmlxC6bjyebsGnr/26IxYKhFFdC67Q7HE0jhDrsrEfxfJMFnsk2zSdt5ofwm2Z1xNhdBg2kfFCzdodI7aHFEdUQ6fddoTioTSPu9zzU9XqBr7Ra+5mTaIwp10heZmlXIjWfu8220ef/tZQ8QSmDX1GSqRLBjUJspesff8Nv9pkQAK3Nvp8YFHKJoFNkSV7QJW7mi/liHYq6DxYqhWk977WYGwzhHA003HNV4OUWhTLJGiPsiFhAlcJVbnVMn6ldnsSauT4unjXA9VBIzaYtSJc29UJYmWyin3MxPz";
+
+        parameters.cameraDirection = VuforiaLocalizer.CameraDirection.FRONT;
+        //this.vuforia = ClassFactory.createVuforiaLocalizer(parameters);
+
+        //relicTemplate.setName("relicVuMarkTemplate");
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java]: #Fix remove commented out code 73-81

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java [hunk 14]
```diff
@@ -0,0 +1,288 @@
...
+        leftArm.setPosition(0.85);
+        // position
+
+        state = States.TIME;
+        isFinished = false;
+
+        //Starts the timer WORKING
+        time.reset();
+
+        //relicTrackables.activate();
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java]: #Fix remove

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java [hunk 15]
```diff
@@ -0,0 +1,288 @@
...
+
+    }
+        @Override
+        public void loop(){
+        currentSeconds = time.seconds();
+            //telemetry.addData("Pitch:", drive.getPitch());
+            //telemetry.update();
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java [hunk 16]
```diff
@@ -0,0 +1,288 @@
...
+            telemetry.update();
+            switch(state){
+                /*case SCAN:
+                    //Scans the pictograph to get correct column
+                    vuMark = RelicRecoveryVuMark.from(relicTemplate);
+                    if(vuMark != RelicRecoveryVuMark.UNKNOWN){
+                        telemetry.addData("Column:", vuMark);
+                        telemetry.update();
+                        state = States.TIME;
+                    } break;*/
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java [hunk 17]
```diff
@@ -0,0 +1,288 @@
...
+
+                /*case DRIVEZONE:
+                    //Drives into the parking zone. UNTESTED/DEACTIVATED
+                    if(!isFinished){
+                        isFinished = drive.encoderDrive(DriveTrain.Direction.S, 0.25, 15);
+                    } else{
+                        isFinished = false;
+                        state = States.END; //TURNBOX
+                    } break;*/
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R1.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R2.java [hunk 18]
```diff
@@ -0,0 +1,284 @@
...
+
+    ColorSensor colorSensorRight;
+    DistanceSensor distanceSensorRight;
+
+    //OpenGLMatrix lastLocation = null;
+    //VuforiaLocalizer vuforia;
+
+    //VuforiaTrackables relicTrackables = this.vuforia.loadTrackablesFromAsset("RelicVuMark");
+    //VuforiaTrackable relicTemplate = relicTrackables.get(0);
+    //RelicRecoveryVuMark vuMark;
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R2.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R2.java [hunk 19]
```diff
@@ -0,0 +1,284 @@
...
+
+        int cameraMonitorViewId = hardwareMap.appContext.getResources().getIdentifier("cameraMonitorViewId", "id", hardwareMap.appContext.getPackageName());
+        VuforiaLocalizer.Parameters parameters = new VuforiaLocalizer.Parameters(cameraMonitorViewId);
+
+        parameters.vuforiaLicenseKey = "AUkgO0T/////AAAAGaYeMjdF+Us8tdP9fJcRhP9239Bwgzo0STjrR4II0s58wT/ja6GlSAQi/ptpHERhBhdNq8MMmlxC6bjyebsGnr/26IxYKhFFdC67Q7HE0jhDrsrEfxfJMFnsk2zSdt5ofwm2Z1xNhdBg2kfFCzdodI7aHFEdUQ6fddoTioTSPu9zzU9XqBr7Ra+5mTaIwp10heZmlXIjWfu8220ef/tZQ8QSmDX1GSqRLBjUJspesff8Nv9pkQAK3Nvp8YFHKJoFNkSV7QJW7mi/liHYq6DxYqhWk977WYGwzhHA003HNV4OUWhTLJGiPsiFhAlcJVbnVMn6ldnsSauT4unjXA9VBIzaYtSJc29UJYmWyin3MxPz";
+
+        parameters.cameraDirection = VuforiaLocalizer.CameraDirection.FRONT;
+        //this.vuforia = ClassFactory.createVuforiaLocalizer(parameters);
+
+        //relicTemplate.setName("relicVuMarkTemplate");
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R2.java]: #Fix remove lines 73-81

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R2.java [hunk 20]
```diff
@@ -0,0 +1,284 @@
...
+
+        //relicTrackables.activate();
+        time.reset();
+
+    }
+        @Override
+        public void loop(){
+        currentSeconds = time.seconds();
+            //telemetry.addData("Pitch:", drive.getPitch());
+            //telemetry.update();
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R2.java]: #Fix commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R2.java [hunk 21]
```diff
@@ -0,0 +1,284 @@
...
+            telemetry.update();
+            switch(state){
+                /*case SCAN:
+                    //Scans the pictograph to get correct column
+                    vuMark = RelicRecoveryVuMark.from(relicTemplate);
+                    if(vuMark != RelicRecoveryVuMark.UNKNOWN){
+                        telemetry.addData("Column:", vuMark);
+                        telemetry.update();
+                        state = States.TIME;
+                    } break;*/
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R2.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R2.java [hunk 22]
```diff
@@ -0,0 +1,284 @@
...
+
+                /*case DRIVEZONE:
+                    //Drives into the parking zone. UNTESTED/DEACTIVATED
+                    if(!isFinished){
+                        isFinished = drive.encoderDrive(DriveTrain.Direction.S, 0.25, 15);
+                    } else{
+                        isFinished = false;
+                        state = States.END; //TURNBOX
+                    } break;*/
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/Red/Auto_R2.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Robot/SubSystems/BlockLift.java [hunk 23]
```diff
@@ -67,10 +77,59 @@ public void slideHeight(boolean dpadUp, boolean dpadDown, Telemetry telemetry) {
...
+        liftMotor.setTargetPosition(liftMotor.getCurrentPosition() - height);
+    }
+
+    public void clamp(boolean a, boolean x, boolean y, boolean b) {
+        if (a) {
+            bottomRightPosition = 0.2;
+            bottomLeftPosition = 0.55;
+//            if (!bottomButtonPress) {
+//                setLift(500);
+//            }
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Robot/SubSystems/BlockLift.java]: #Fix remove commented out code

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Robot/SubSystems/BlockLift.java [hunk 24]
```diff
@@ -67,10 +77,59 @@ public void slideHeight(boolean dpadUp, boolean dpadDown, Telemetry telemetry) {
...
+
+        if (y) {
+            topLeftPosition = 0.37;
+            topRightPosition = 0.26;
+        } else if (b) {
+            topLeftPosition = 0;
+            topRightPosition = 0.65;
+//            if (!bottomButtonPress) {
+//                setLift(500);
+//            }
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Robot/SubSystems/BlockLift.java]: #Fix remove commented out code

**Review by jschool504 — CHANGES_REQUESTED** [review]

**Comment by jschool504** [comment]
#FYI @amoller655 @Andor16 We should look at moving the case definitions into a common file and trying to collect case implementations into a common file so that individual autonomous programs can be slimmed down to defining what cases are executed in what order for a given autonomous program.

**Review by amoller655 — APPROVED** [review]
I'm fine with all of these changes and I don't see any problems other than what Jeremy has addressed. Thanks for streamlining my code as well!

**Comment by amoller655** [comment]
Not to say that Jeremy's issues don't need to be fixed, it's just that I don't see anything other than that that that needs to be fixed

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/States.java [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+    DRIVEBACK,
+    STRAFE,
+    END,
+    GRAB,
+    DROP,
+    LIFT,
+    CLOSE,
+    DRIVEFORWARD,
+    OPEN,
+    DRIVEBACK2,
```
Comment (jschool504) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/GMR/Autonomous/States.java]: #fix We need a more descriptive name for this state.

**Review by jschool504 — APPROVED** [review]

**Comment by jschool504** [comment]
LGTM after last comment fixed 👍

**Pull request merged**